code/krepel/system/ifile.d

```diff
@@ -0,0 +1,104 @@
+module krepel.system.ifile;
+
+import krepel.memory;
+import krepel.container;
+import krepel.meta;
+
+@nogc:
+
+enum FileOpenMode
+{
+  Read = 0x1,
+  Write = 0x2
+}
+
+interface IFile
+{
+  @nogc:
+
+  /// Reads data from a file
+  /// Params:
+  /// Region = Where the data will be written into which was red from the file
+  /// Returns: Returns the amount of bytes written into Region. The value is >= 0 and <= Region.length.
+  long Read(void[] Region)
+  in
+  {
+    assert(Region.length > 0);
+  }
+  out(result)
+  {
+    assert(result >= 0);
+  }
+
+  /// Reads data from a file until it finds a \r or \n or the eof
+  /// Params:
+  /// Array = Where the data will be written into which was red from the file
+  /// Returns: Returns the amount of bytes written into Region. The value is >= 0 and <= Region.length.
+  long ReadLine(CharType)(ref Array!CharType Array)
+    if(IsSomeChar!CharType)
+  out(result)
+  {
+    assert(result >= 0);
+  }
+  body
+  {
+    CharType[10] Buffer;
+    while(true)
+    {
+      auto CurPosition = MoveCursor(0);
+      auto ReadCount = Read(cast(ubyte[])Buffer);
```

Comment:
I suppose this should be `cast(void[])` now, after the removal of `MemoryRegion`.